django/middleware/csrf.py

```diff
@@ -167,22 +186,31 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
                 if referer.scheme != 'https':
                     return self._reject(request, REASON_INSECURE_REFERER)
 
+                good_referers = []
```

Comment:
This looks odd. If `CSRF_USE_SESSIONS`, then the `CSRF_COOKIE_DOMAIN` should be irrelevant, and `SESSION_COOKIE_DOMAIN` should be used instead. We shouldn't get more good referrers then we had before.